File: backend/src/main/java/com/bidbridge/backend/entity/Port.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "ports", schema = "public")
public class Port {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

    private Instant createdAt;
}

File: frontend/src/types/port.ts
export interface Port {
  id: string;
  name: string;
  company_id: string;
  company_name: string;
  created_at: string;
}

export interface PortApiResponse {
  id: string;
  name: string;
  companyId: string;
  companyName: string;
  createdAt: string;
}

export function transformPortFromApi(apiPort: PortApiResponse): Port {
  return {
    id: apiPort.id,
    name: apiPort.name,
    company_id: apiPort.companyId,
    company_name: apiPort.companyName,
    created_at: apiPort.createdAt,
  };
}
